joomla/joomla-cms
# [4.0] Set default_view base on component name

Pull Request for Issue # .

### Summary of Changes
In Joomla 3, when we don't set value for $default_view property of controller, Joomla will parse class name of controller and use it as default_view (see https://github.com/joomla/joomla-cms/blob/staging/libraries/src/MVC/Controller/BaseController.php#L427 and https://github.com/joomla/joomla-cms/blob/staging/libraries/src/MVC/Controller/BaseController.php#L767)

In Joomla 4, we use namespace for controller class, so the method getName() doesn't return name of component anymore, thus we had to set value manually for $default_value property of every DisplayController

This PR modify code a bit, set default_view property for controller base on component name, and with this small change, we don't have set default_view property anymore (in case default view is the same with component name) like in Joomla 3.


### Testing Instructions
1. Install Joomla 4.

2. Apply patch

3. As the change is the same, you only need to test com_banners. Access to Components -> Banners -> Banners and confirm that it is still working as before.
